packages/nango-yaml/lib/parser.v2.ts

```diff
@@ -0,0 +1,261 @@
+                    webhookSubscriptions = sync['webhook-subscriptions'];
+                } else {
+                    webhookSubscriptions = [sync['webhook-subscriptions']];
+                }
+            }
+
+            const parsedSync: ParsedNangoSync = {
+                name: syncName,
+                type: 'sync',
+                description: sync.description || '',
+                sync_type: sync.sync_type === 'incremental' ? 'incremental' : 'full',
+                usedModels: Array.from(modelNames),
+                runs: sync.runs,
+                track_deletes: sync.track_deletes || false,
+                auto_start: sync.auto_start === false ? false : true,
+                input: modelInput?.name || null,
+                output: modelOutput.map((m) => m.name),
+                scopes: Array.isArray(sync.scopes) ? sync.scopes : sync.scopes ? sync.scopes.split(',') : [],
+                endpoints,
+                webhookSubscriptions
+            };
+
+            parsedSyncs.push(parsedSync);
+        }
+
+        return parsedSyncs;
+    }
+
+    parseActions({
+        actions,
+        usedModels,
+        integrationName
+    }: {
+        actions: NangoYamlV2Integration['actions'];
+        usedModels: Set<string>;
+        integrationName: string;
+    }): ParsedNangoAction[] {
+        const parsedActions: ParsedNangoAction[] = [];
+
+        for (const actionName in actions) {
+            const action = actions[actionName];
+            if (!action) {
+                continue;
+            }
+
+            const modelNames = new Set<string>();
+
+            const modelOutput = this.getModelForOutput({ rawOutput: action.output, usedModels, name: actionName, type: 'action', integrationName });
+            if (modelOutput && modelOutput.length > 0) {
+                modelOutput.forEach((m) => modelNames.add(m.name));
+            }
+
+            const modelInput = this.getModelForInput({ usedModels, rawInput: action.input, name: actionName, type: 'action', integrationName });
+            if (modelInput) {
+                modelNames.add(modelInput.name);
+            }
+
+            const endpoint: NangoSyncEndpoint = {};
+            if (action.endpoint) {
+                const split = action.endpoint.split(' ') as [HTTP_VERB, string];
+
+                if (this.endpoints.has(action.endpoint)) {
+                    this.errors.push(new ParserErrorDuplicateEndpoint({ endpoint: action.endpoint, path: [integrationName, 'syncs', actionName] }));
+                    continue;
+                }
+
+                this.endpoints.add(action.endpoint);
+                endpoint[split[0]] = split[1];
+            }
+
+            const parsedAction: ParsedNangoAction = {
+                name: actionName,
+                type: 'action',
+                description: action.description || '',
+                scopes: Array.isArray(action.scopes) ? action.scopes : action.scopes ? action.scopes.split(',') : [],
```

Comment:
I found double ternary super confusing. That might just be me though

Reply:
true, changing